FIS2425/authorization-svc
# Feature/19 2fa

### Description

This PR adds 2FA to the login process for enhanced security.

### Changes made

- Updated login process to handle 2FA-enabled accounts by validating credentials and creating a pending session in Redis.
- Added an endpoint to enable 2FA.
- Added an endpoint to verify 2FA which generates JWT tokens in case of a successful verification.

### Steps to test

1. Log in with a user.
2. Enable 2FA for a user via the `/users/enable-2fa` endpoint and scan the QR code with an authenticator app.
3. Log out and try to log in again. This time, you'll need to send the TOTP code within another request, described in the next step.
4. Call `/users/verify-2fa` with the TOTP code generated by the authenticator app.
5. If the code is valid, you will have successfully logged in using 2FA.

**Note:** remember to update your `.env` file with the new `TOTP_EXPIRATION` variable.
